File: ui/moai/src/lobechat/types/aiProvider.ts
import { AiModelForSelect, EnabledAiModel, ModelSearchImplementType } from './aiModel';
import { SmoothingParams } from './llm';

export const AiProviderSourceEnum = {
  Builtin: 'builtin',
  Custom: 'custom',
} as const;
export type AiProviderSourceType = (typeof AiProviderSourceEnum)[keyof typeof AiProviderSourceEnum];

/**
 * only when provider use different sdk
 * we will add a type
 */
export const AiProviderSDKEnum = {
  Anthropic: 'anthropic',
  Azure: 'azure',
  AzureAI: 'azureai',
  Bedrock: 'bedrock',
  Cloudflare: 'cloudflare',
  Google: 'google',
  Huggingface: 'huggingface',
  Ollama: 'ollama',
  Openai: 'openai',
  Volcengine: 'volcengine',
} as const;

export type AiProviderSDKType = (typeof AiProviderSDKEnum)[keyof typeof AiProviderSDKEnum];

export interface AiProviderSettings {
  /**
   * whether provider show browser request option by default
   *
   * @default false
   */
  defaultShowBrowserRequest?: boolean;
  /**
   * some provider server like stepfun and aliyun don't support browser request,
   * So we should disable it
   *
   * @default false
   */
  disableBrowserRequest?: boolean;
  /**
   * whether provider support edit model
   *
   * @default true
   */
  modelEditable?: boolean;

  proxyUrl?:
    | {
        desc?: string;
        placeholder: string;
        title?: string;
      }
    | false;

  /**
   * default openai
   */
  sdkType?: AiProviderSDKType;
  searchMode?: ModelSearchImplementType;
  showAddNewModel?: boolean;
  /**
   * whether show api key in the provider config
   * so provider like ollama don't need api key field
   */
  showApiKey?: boolean;
  /**
   * whether show checker in the provider config
   */
  showChecker?: boolean;
  showDeployName?: boolean;
  showModelFetcher?: boolean;
  /**
   * whether to smoothing the output
   */
  smoothing?: SmoothingParams;
}

export interface AiProviderListItem {
  description?: string;
  enabled: boolean;
  id: string;
  logo?: string;
  name?: string;
  sort?: number;
  source: AiProviderSourceType;
}

// Detail Query

export interface AiProviderCard {
  /**
   * the default model that used for connection check
   */
  checkModel?: string;
  config: AiProviderSettings;
  description?: string;
  enabled: boolean;
  enabledChatModels: string[];
  /**
   * provider's website url
   */
  homeUrl?: string;
  id: string;
  logo?: string;
  /**
   * the url show the all models in the provider
   */
  modelsUrl?: string;
  /**
   * the name show for end user
   */
  name: string;
}

export interface AiProviderDetailItem {
  /**
   * the default model that used for connection check
   */
  checkModel?: string;
  description?: string;
  enabled: boolean;
  fetchOnClient?: boolean;
  /**
   * provider's website url
   */
  homeUrl?: string;
  id: string;
  keyVaults?: Record<string, any>;
  logo?: string;
  /**
   * the url show the all models in the provider
   */
  modelsUrl?: string;
  /**
   * the name show for end user
   */
  name: string;
  settings: AiProviderSettings;
  source: AiProviderSourceType;
}

export interface AiProviderSortMap {
  id: string;
  sort: number;
}

// --------

export interface EnabledProvider {
  id: string;
  logo?: string;
  name?: string;
  source: AiProviderSourceType;
}

export interface EnabledProviderWithModels {
  children: AiModelForSelect[];
  id: string;
  logo?: string;
  name: string;
  source: AiProviderSourceType;
}

export interface AiProviderRuntimeConfig {
  fetchOnClient?: boolean;
  keyVaults: Record<string, string>;
  settings: AiProviderSettings;
}

export interface AiProviderRuntimeState {
  enabledAiModels: EnabledAiModel[];
  enabledAiProviders: EnabledProvider[];
  runtimeConfig: Record<string, AiProviderRuntimeConfig>;
}

File: src/ai/MoAI.AI.Shared/Models/AiProvider.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace MoAI.AI.Models;

/// <summary>
/// Ai 接口类型，不区分厂家.
/// </summary>
public enum AiProvider
{
    /// <summary>
    /// 自定义.
    /// </summary>
    [JsonPropertyName("custom")]
    [EnumMember(Value = "custom")]
    Custom,

    /// <summary>
    /// Anthropic.
    /// </summary>
    [JsonPropertyName("anthropic")]
    [EnumMember(Value = "anthropic")]
    Anthropic,

    /// <summary>
    /// Azure, Azure == AzureAI.
    /// </summary>
    [JsonPropertyName("azure")]
    [EnumMember(Value = "azure")]
    Azure,

    /// <summary>
    /// Google.
    /// </summary>
    [JsonPropertyName("google")]
    [EnumMember(Value = "google")]
    Google,

    /// <summary>
    /// HuggingFace.
    /// </summary>
    [JsonPropertyName("huggingface")]
    [EnumMember(Value = "huggingface")]
    Huggingface,

    /// <summary>
    /// Mistral.
    /// </summary>
    [JsonPropertyName("mistral")]
    [EnumMember(Value = "mistral")]
    Mistral,

    /// <summary>
    /// Ollama.
    /// </summary>
    [JsonPropertyName("ollama")]
    [EnumMember(Value = "ollama")]
    Ollama,

    /// <summary>
    /// OpenAI.
    /// </summary>
    [JsonPropertyName(name: "openai")]
    [EnumMember(Value = "openai")]
    Openai
}
